### ISSUE TITLE
How do I create my own training data?

### ISSUE THREAD

**Issue by off99555** [sebastianstarke/AI4Animation#12]
There are 3 questions about SIGGRAPH 2017 paper:
1. How do I create my own training data, given that I have VR devices like leg bands, hand controller, and headset?
2. Is it possible to include hand and head input from VR devices into the model too (instead of only Up/Down/Left/Right input)?
3. What is the training set size for the model provided in this repository?

I will need to read more about the paper. Any advice is appreciated. Thank you!

**Comment by keicoon** [comment]
1. Get bone based hierchy position and rotation by VR device. And make training data by follow BVH or your custom animation format.
2. You can make your phase-function not only "foot" to "hand and head input". Just modify foot-phase part to custom-phase and training by VR traning data set instead of pfnn-data.
3. Get training data set in [here](https://github.com/ShikamaruZhang/PFNN).

**Comment by off99555** [comment]
I will try figuring this out and if I encounter problems I will reopen this issue again.

**Comment by sebastianstarke** [comment]
Sorry for the late response. It might be hard applying the original 2017 work to any domain outside locomotion. This is because the design of the phase is difficult if no clear rules are available, and will also blur motions of different modalities (i.e. walk, grasp, sit). Anyways, please let me know if you have further questions.

**Comment by Keneyr** [comment]
Not sure if you can see this comment. I have seen the codes of  Siggraph_2017 paper.  I tried to make my own  data ----I used the .bvh files from CMU where the skeleton has 57 bones . Not the 31 bones in your Demo. I used the MotionExporter.cs to export the bvh to input.txt etc. Since the bvh file consists of a simple sequence of walking,  the txt file is  only
![image](https://user-images.githubusercontent.com/25053570/53476827-43ef7d80-3aae-11e9-91e1-6d6f5ec7864f.png)
but your txt in 2018Demo is 
![image](https://user-images.githubusercontent.com/25053570/53476846-523d9980-3aae-11e9-855e-bc0e297cbeb2.png)
Can you tell me why,why your txt file is so big ,but mine is so small...... what should I do to get the right data?

**Comment by sebastianstarke** [comment]
I think you will need to go through the whole data processing pipeline. Start with loading the .bvh into the MotionEditor, assign all the labels you need, and in the MotionExporter, adjust your output format. It's very likely that things need to changed from the quadruped data setup to human data setup. It seems like there is almost no content in each data sample, or the amount of data used is small?

**Comment by Keneyr** [comment]
yes, I loaded the .bvh into the MotionEditor, and exported the first .bvh file to txt file .
![image](https://user-images.githubusercontent.com/25053570/53491005-2c28f100-3ad0-11e9-900b-1225ae2e67c0.png)
![image](https://user-images.githubusercontent.com/25053570/53491014-321ed200-3ad0-11e9-95f3-a07959e8459a.png)
the 02_01.bvh is only 439KB, so there is no doubt that the corresponding txt file is small 
In your 2017paper ,you said that  Once finished we have around 1hour of raw motion capture data captured at 60fps which constitutes around 1.5GB of data.
So your whole 1.5GB data is all  in one file , or you just separated the data in several files?

**Comment by sebastianstarke** [comment]
It all comes together in one file, basically the exporter does batch exporting and writes all together because of computing normalisation values. You can then load the input and output into your neural network. That's the pipeline basically.

**Comment by Keneyr** [comment]
There it is ! Thank you very much ! I think I got it.
